src/components/views/rooms/RoomSettings.js

```diff
@@ -252,6 +270,30 @@ module.exports = React.createClass({
         });
     },
     
+    _onRoomAccessRadioToggle: function(ev) {
+        var self = this;
+        switch (ev.target.value) {
+            case "invite_only":
+                self.setState({
+                    join_rule: "invite",
+                    guest_access: "can_join",
```

Comment:
I would love to see a 2x2 matrix of the options here:
```
                      join_rule
                  INVITE  |  PUBLIC
       -------------------+----------------
guest  CAN_JOIN | inv_only| pub_with_guest
access -------------------+----------------
       NO_JOIN  | inv_only| pub_no_guest
       -------------------+----------------
```

You can see my concern more clearly here: there's no way to actually set all 4 states with the present UI, and no option seems intuitive here I think.